### PR TITLE
Generic code syntax fixes/improvements

### PR DESCRIPTION
**Pull request by funder7**
Hi, I started this update due to some wrong html markup that I've found while doing some changes. 
While doing these checks I noticed that some parts of the code can be updated to use better syntax (clearer and sometimes faster).
In particular, boolean operations should migrate to type safe checks, as soon or later unsafe checks will be declared completely unsupported. 

There's still much work to do, I hope that you will find this useful. 

Cheers 

## Issue Reference
#5361 

## Description

- safely use `===` with string comparison
- ternary operator always return a boolean
- `array_search`: better specify if strict mode
- `$arr[] = $val;` is 2x faster than array_push
- fixed `</br>` typo (should be `<br/>` or `<br>`)
- fixed some `<span>` tags not closed
- fixed some `<div>` tafs not closed
- replaced `event.keyCode` (deprecated) @see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#browser_compatibility
- blade: last statement doesn't need semicolon ;
- use `??` operator instead of isset (where possible)
- type="text/css" is obsolete in `<style>` tag
- php `is_object()` shorter than `gettype() === 'object'`
- `[^0-9,.]` < no need to escape comma & dot
- php type casting in ternary operator should be applied to both results
- wrap wanted binary condition with parentheses to avoid wrong interpretation
- `CSS` 
    - unit of measure for `0px` is redundant
    - `background-image: URL(..)` ← use lowercase
    - `::last-child` → `:last-child` (pseudo class !selector)
    - some properties are overwritten (search 'is overwritten')
    - use shorthand (margin, padding, radius..)
    - `:first/last-child()` <- parentheses not required
- `datagrid-plus.vue`: moved button outside of list (check) + missing checkbox id (4 label)
- many duplicated translation keys (until italian)
- updated italian translation for Admin package

## How To Test This?
1. clone the project
2. run every package laravel mix
3. clear config & views
4. test  

## Documentation
- [ ] My pull request requires an update on the documentation repository.

### CODE REVIEW

**Comment by devansh-webkul** [comment]
I just checked the code but still for full review you need to pass all the test cases.

It looks like you have added a strict check to all over the place.

**Comment by funder7** [comment]
Hi @devansh-webkul, yes I've added strict checking but only where there return type is of one type only (e.g. string comparison). I left untouched the comparison where the return type can be of 2 or more types.

I will check why tests are failing in the next days. Thanks.

**Comment by funder7** [comment]
@devansh-webkul tests are passing now! 
[Here](http://sandbox.onlinephpfunctions.com/code/90a512d62882d43a0b3ee7359ef0a610b792e577) you can find a brief explanation on the issue.

If the value to check may be `null` or empty string, php `empty($var)` can be used safely, as it behaves exactly in the same way as `$var == null` (true), or `$var == ''` (true). Previously I updated the code to `$var === ''` which return false if `$var` is null. 

The test failing was `\Tests\Unit\CartRule\CartRuleCest`, which sets the app default country to `DE` in the second line. I suspect that `DE` is not valid for test. 

Basically switching to strict type checking, made the `\Webkul\Tax\Helpers\Tax::__construct` load the `app.default_country`. I've read the config just now, and noticed that `default_country` is set to null in the base project (not being loaded from env variables, nor having a default value). 

`default-location-calculation.country` in Tax package is set to empty string instead. 

I can tell you that this was the culprit which caused the previous test to fail; Maybe it needs a proper check. 


Cheers :-)

**Comment by devansh-webkul** [comment]
@funder7,

This PR is too huge, I will see this in the end when all remaining PRs get merged.

**Comment by devansh-webkul** [comment]
Meanwhile please resolve the conflicts.

**Comment by funder7** [comment]
Conflicts solved. 

Probably it's done on purpose, anyway I've found that build artifacts should be not tracked in the repository. It's a good habit in general to keep source code only in the repository.

Cheers

File: packages/Webkul/Admin/src/Resources/views/sales/invoices/create.blade.php [hunk 1]
```diff
@@ -246,7 +246,7 @@
                                                             <div class="item-options">
 
                                                                 @foreach ($item->additional['attributes'] as $attribute)
-                                                                    <b>{{ $attribute['attribute_name'] }} : </b>{{ $attribute['option_label'] }}</br>
```
Comment (funder7) [inline packages/Webkul/Admin/src/Resources/views/sales/invoices/create.blade.php:249]: </br> is not a valid html tag, replace with <br/>

File: packages/Webkul/Admin/src/Resources/views/tax/tax-rates/index.blade.php [hunk 2]
```diff
@@ -52,7 +52,7 @@
                 @csrf()
                 <div class="control-group" :class="[errors.has('file') ? 'has-error' : '']">
                     <label for="file" class="required">{{ __('admin::app.export.file') }}</label>
-                    <input v-validate="'required'" type="file" class="control" id="file" name="file" data-vv-as="&quot;{{ __('admin::app.export.file') }}&quot;" value="{{ old('file') }}"/ style="padding-top: 5px">
```
Comment (funder7) [inline packages/Webkul/Admin/src/Resources/views/tax/tax-rates/index.blade.php:55]: typo

File: packages/Webkul/Shop/src/Resources/views/checkout/cart/index.blade.php [hunk 3]
```diff
@@ -87,12 +87,12 @@
 
                                             @auth('customer')
                                                     @php
-                                                        $showWishlist = core()->getConfigData('general.content.shop.wishlist_option') == "1" ? true : false;
```
Comment (funder7) [inline packages/Webkul/Shop/src/Resources/views/checkout/cart/index.blade.php:90]: no need to use ternary here, condition is already returning a boolean

File: packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php [hunk 4]
```diff
@@ -6,7 +6,7 @@
         <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
 
         {{-- lang supports inclusion --}}
-        <style type="text/css">
```
Comment (funder7) [inline packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php:9]: deprecated attribute value (text/css is the default)

File: packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
                 foreach (app('Webkul\Category\Repositories\CategoryRepository')->getVisibleCategoryTree(core()->getCurrentChannel()->root_category_id) as $category){
                     if ($category->slug)
-                        array_push($categories, $category);
+                        $categories[] = $category;
```
Comment (funder7) [inline packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php:10]: - cleaner; 
- faster;

[more here](https://stackoverflow.com/questions/559844/which-is-faster-in-php-array-value-or-array-pusharray-value)

File: packages/Webkul/Shop/src/Resources/views/layouts/header/index.blade.php [hunk 6]
```diff
@@ -40,7 +40,7 @@ class="search-field"
 
                         <div class="search-icon-wrapper">
 
-                            <button class="" class="background: none;" aria-label="Search">
```
Comment (funder7) [inline packages/Webkul/Shop/src/Resources/views/layouts/header/index.blade.php:43]: duplicate

File: packages/Webkul/Shop/src/Resources/views/layouts/header/index.blade.php [hunk 7]
```diff
@@ -77,7 +77,7 @@ class="search-field"
                             <i class="icon compare-icon"></i>
                             <span class="name">
                                 {{ __('shop::app.customer.compare.text') }}
-                                <span class="count">(<span id="compare-items-count"></span>)<span class="count">
```
Comment (funder7) [inline packages/Webkul/Shop/src/Resources/views/layouts/header/index.blade.php:80]: !important 
Opening tag is duplicated into closing tag.
I've tested the outcome of the fixed version, seems ok to me.

File: packages/Webkul/Velocity/src/Resources/views/admin/content/create.blade.php [hunk 8]
```diff
@@ -49,7 +49,7 @@
 
                         <div class="control-group" :class="[errors.has('position') ? 'has-error' : '']">
                             <label for="position" class="required">
-                                {{ __('velocity::app.admin.contents.page.position') }}</span>
```
Comment (funder7) [inline packages/Webkul/Velocity/src/Resources/views/admin/content/create.blade.php:52]: "span" never opened

File: packages/Webkul/Velocity/src/Resources/views/admin/content/edit.blade.php [hunk 9]
```diff
@@ -62,14 +62,14 @@
...
                             </label>
-                            <input type="text" v-validate="'required|max:100'" class="control" id="title" name="{{$locale}}[title]" value="{{ old($locale)['title'] ?? isset($translation->title) ? $translation->title : '' }}" data-vv-as="&quot;{{ __('velocity::app.admin.contents.page.title') }}&quot;"/>
+                            <input type="text" v-validate="'required|max:100'" class="control" id="title" name="{{$locale}}[title]" value="{{ old($locale)['title'] ?? $translation->title ?? '' }}" data-vv-as="&quot;{{ __('velocity::app.admin.contents.page.title') }}&quot;"/>
 
                             <span class="control-error" v-if="errors.has('{{$locale}}[title]')" v-text="errors.first('{!!$locale!!}[title]')"></span>
                         </div>
 
                         <div class="control-group" :class="[errors.has('position') ? 'has-error' : '']">
                             <label for="position" class="required">
-                                {{ __('velocity::app.admin.contents.page.position') }}</span>
```
Comment (funder7) [inline packages/Webkul/Velocity/src/Resources/views/admin/content/edit.blade.php:72]: `</span>` never opened

File: packages/Webkul/Velocity/src/Resources/views/shop/customers/account/partials/sidemenu.blade.php [hunk 10]
```diff
@@ -61,7 +61,7 @@
                 <li class="{{ $menu->getActive($subMenuItem) }}" title="{{ trans($subMenuItem['name']) }}">
                     <a class="unset fw6 full-width" href="{{ $subMenuItem['url'] }}">
                         <i class="icon {{ $index }} text-down-3"></i>
-                        <span>{{ trans($subMenuItem['name']) }}<span>
```
Comment (funder7) [inline packages/Webkul/Velocity/src/Resources/views/shop/customers/account/partials/sidemenu.blade.php:64]: `<span>` instead of `</span>`

File: packages/Webkul/Velocity/src/Resources/views/shop/home/new-products.blade.php [hunk 11]
```diff
@@ -12,7 +12,7 @@
     product-title="{{ __('shop::app.home.new-products') }}"
     product-route="{{ route('velocity.category.details', ['category-slug' => 'new-products', 'count' => $count]) }}"
     locale-direction="{{ $direction }}"
-    show-recently-viewed="{{ (Boolean) $showRecentlyViewed ? 'true' : 'false' }}"
+    show-recently-viewed="{{ (Boolean) ($showRecentlyViewed ? 'true' : 'false') }}"
```
Comment (funder7) [inline packages/Webkul/Velocity/src/Resources/views/shop/home/new-products.blade.php:15]: probably a `$showRecentlyViewed === 'true' || $showRecentlyViewed === true` would fit better here (without type casting)

File: packages/Webkul/Velocity/src/Resources/views/shop/products/reviews/index.blade.php [hunk 12]
```diff
@@ -5,9 +5,8 @@
 @endsection
 
 @php
-    $ratings = [
-        '', '', '', ''
-    ];
+    // immediately overridden (check & remove) :
+    // $ratings = [ '', '', '', '' ];
 
     $ratings = [
```
Comment (funder7) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/reviews/index.blade.php:11]: overrides previous declaration, can be removed

**Comment by devansh-webkul** [comment]
Same here also,
https://github.com/bagisto/bagisto/pull/5401#issuecomment-993147621

**Comment by devansh-webkul** [comment]
Hi @funder7,

I think PR is getting too old, I am closing both pull requests. 

You can hit a new PR from the same branch once done.
